src/map/ai/controllers/trust_controller.cpp

```diff
@@ -95,14 +102,49 @@ void CTrustController::DoCombatTick(time_point tick)
                 }
             }
         }
+		
+		POwner->PAI->EventHandler.triggerListener("TRUST_COMBAT_TICK", POwner, POwner->PMaster, PTarget);
+		luautils::OnMobFight(POwner, PTarget);
     }
 }
 
 void CTrustController::DoRoamTick(time_point tick)
 {
     if (POwner->PMaster->PAI->IsEngaged())
     {
-        POwner->PAI->Internal_Engage(POwner->PMaster->GetBattleTargetID());
+        CCharEntity* PChar = (CCharEntity*)POwner->PMaster;
+
+        // Only engage the mob if the player has hate
+        for (SpawnIDList_t::iterator it = PChar->SpawnMOBList.begin(); it != PChar->SpawnMOBList.end(); ++it)
```

Comment:
Trusts do a little more than just check for hate for engagement. You have to actually perform an attack on the mob to get it to engage, doesn't matter if you hit or miss. After it has successfully engaged, then you can target lock off the mob and the trust will continue the attack up until the point where: 1 Trust dies, 2 Mob dies, or 3 Player Disengages. Here is some Video of different Targeting Mechanics for the Trusts and how they work:
https://youtu.be/0HIBwXu_BvI